test/libsolidity/util/TestFunctionCall.cpp

```diff
@@ -125,88 +174,173 @@ string TestFunctionCall::format(
 string TestFunctionCall::formatBytesParameters(
 	ErrorReporter& _errorReporter,
 	bytes const& _bytes,
-	dev::solidity::test::ParameterList const& _params
+	string const& _signature,
+	dev::solidity::test::ParameterList const& _params,
+	bool _highlight
 ) const
 {
-	stringstream resultStream;
+	using ParameterList = dev::solidity::test::ParameterList;
+
+	stringstream os;
+	string functionName{_signature.substr(0, _signature.find("("))};
 
-	if (_bytes.empty())
-		return {};
 	auto sizeFold = [](size_t const _a, Parameter const& _b) { return _a + _b.abiType.size; };
 	size_t encodingSize = std::accumulate(_params.begin(), _params.end(), size_t{0}, sizeFold);
 
-	if (encodingSize != _bytes.size())
+	/// Infer type from Contract ABI. Used to generate values for
+	/// auto-correction during interactive update routine.
+	ParameterList abiParams;
+	for (auto const& function: m_contractABI)
+		if (function["name"] == functionName)
+			for (auto const& output: function["outputs"])
+			{
+				auto types = contractABITypes(output["type"].asString());
+				for (auto const& type: types)
+					abiParams.push_back(Parameter{bytes(), "", type, FormatInfo{}});
+			}
+
+
+	/// If parameter count does not match, take types defined by ABI, but only
+	/// if the contract ABI is defined (needed for format tests where the actual
+	/// result does not matter).
+	ParameterList preferredParams;
+	if (m_contractABI && (_params.size() != abiParams.size()))
+	{
 		_errorReporter.warning(
```

Comment:
Is this error really that important? Won't we display an error already due to the values not matching? Perhaps an assertion can be put here that checks that the sizes are the same if the values are the same.

Reply:
Yes, it would result in an error already. But it turned out, that it makes sense to have this information (expected vs. actual size) when developing more complex tests (e.g. calldata arrays). The idea behind the slightly over-engineered error reporter, which is also part of this PR, was to give the user as many information as needed. I'm thinking about test-driven development, which would also benefit from a verbose output. 

One drawback I can see here, is the mapping from a Solidity type to the test-internal ABIType representation, which is kind of incomplete right now (reference types need to be fixed). But still, you would have the warning even if the auto-corrected value or its size is incorrect.

Reply:
Ok!